docs/cpp_client/cpp_examples.md

```diff
@@ -0,0 +1,229 @@
+in "Play" mode, read in configuration values and create an instance of a 
+`Simulator` object:
+
+```cpp
+//Read JSON files in cpp_client/config directory
+Configuration config(
+    "simulator-cpp-client/config/simulator.json",
+    "simulator-cpp-client/config/vehicle.json",
+    "simulator-cpp-client/config/weather.json",
+    "simulator-cpp-client/config/scenario.json"
+);
+// Default simulator port is 8999
+Simulator& simulator = Simulator::getInstance(config, "127.0.0.1", 8999);
+```
+
+### Creating Sensors
+
+Each sensor has a configuration `struct` that must be setup either by reading in 
+a JSON configuration file or assigning values to the `struct` directly. Here, 
+an example of constructing a camera sensor from JSON should be:
+
+```cpp
+// Read the JSON from file
+std::ifstream in("/path/to/my/config.json");
+nlohmann::json file;
+file << in;
+
+// Import into the camera config struct
+CameraConfig camera_config;
+camera_config.from_json(file);
+```
+
+Alternatively, the `struct` can be setup directly in the code:
+
+```cpp
+CameraConfig camera_config;
+camera_config.server_ip = ip;
+camera_config.listen_port = 8100;
+camera_config.location.z = 200;
+camera_config.rotation.pitch = -5;
+camera_config.resolution = CameraConfig::Resolution(1024,1024);
+```
+
+After the sensor's configuration is created, all that needs to be done is 
+configure the sensor:
+
+```cpp
+Sensor camera(CameraConfig);
+camera.configure();
+```
+
+### Stepping in Replay Mode
+
+After the `simulator` object is connected and the sensors are configured, the 
+simulation can be stepped in replay mode:
+
+```cpp
+for(; idx < nSteps; idx++)
+{	
+    //step simulator
+    task = std::async([&simulator, &idx](){
+        return simulator.step(idx, 1);
+    });
+```
+
+After each step, a sensor sample can be acquired using the sensor's `sample()` 
+function:
+
+```cpp
+    //sample all sensors
+    for(auto& sensor : sensors)
+    {
+        sensor.sample();
+    }
+    if(!task.get()){
+        break;
+    }
+} // for
+```
+
+### Controlling the EGO Vehicle
+
+As an alternative to Replay mode, a closed loop control over the EGO vehicle can 
+be done by sending control commands to the simulator, here the vehicle is 
+requested to move forward with 50% throttle:
+
+```cpp
+simulator.send_command(ApiMessage(123, EgoControl_ID, true, 
+    {   {"forward_amount", 0.5}, 
+        {"right_amount", 0.0},
+        {"brake_amount", 0.0},
+        {"drive_mode", 1}
+    }));
+```
+
+## Replay Example
+
+A full example of controlling the EGO vehicle in Replay mode can be found in 
+`monodrive-client/cpp-client/cpp-examples/replay/replay_example.cpp`. This 
+example demonstrates:
+
+* Configuring and connecting to a running instance of the monoDrive Simulator
+* Configuring and connecting to a Camera sensor (as discussed above)
+* Configuring the Viewport Sensor for the camera on the simulator
+* Stepping the a monoDrive Trajectory File in Replay mode (as discussed above)
```

Comment:
Get rid of the or a like:  Stepping a monoDrive Trajectory File in Replay mode (as discussed above)

Reply:
Fixed